### PR TITLE
Feature/its 63

### PR DESCRIPTION
**Pull request by hamzalatif-e2x**
Basic local firestore emulator setup that runs a simple test against a locally running function

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -37,6 +37,9 @@ jobs:
           npm install
           npm test
 
+      - name: integration test
+        run : |
+          npm run readFromFirestoreTest
```
Comment (aalmansa-e2x) [inline .github/workflows/main.yml:42]: I guess we will be able to have more commands here?

File: firestoreEmulator/teacher1.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "id": "1",
+  "firstName": "Kristin",
+  "lastName": "Smith",
+  "occupation": "Teacher",
```
Comment (aalmansa-e2x) [inline firestoreEmulator/teacher1.json:5]: Should we load some catetories instead?

File: src/index.ts [hunk 3]
```diff
@@ -10,3 +11,14 @@ export const goodbyeWorld: HttpFunction = (req, res) => {
   util();
   res.send('Goodbye, World!');
 };
+
+var firestoreClient = admin.initializeApp({ projectId: "test" }).firestore()
+
+export const readFromFirestore: HttpFunction = async (req, res) => {
```
Comment (aalmansa-e2x) [inline src/index.ts:17]: OK, this will go away when we have the actual test for adding categories

File: .firebaserc [hunk 4]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (aalmansa-e2x) [inline .firebaserc:1]: Is this file needed?

File: package.json [hunk 5]
```diff
@@ -11,32 +11,42 @@
   "keywords": [],
   "scripts": {
     "zip": "bestzip $npm_config_file build/* package.json",
-    "test": "jest",
+    "test": "jest -c ./jest.config.js",
+    "test:integration": "jest -c ./jest.config.integration.js",
```
Comment (aalmansa-e2x) [inline package.json:15]: Nice one!

File: package.json [hunk 6]
```diff
@@ -11,32 +11,42 @@
...
     "prepare": "npm run compile",
-    "pretest": "npm run compile",
     "start": "functions-framework --source=build/src/ --target=helloWorld",
+    "start:readFromFirestore": "export FIRESTORE_EMULATOR_HOST=localhost:8081 && tsc && functions-framework --source=build/src/ --target=readFromFirestore",
+    "stop:function": "kill $(lsof -t -i:8080)",
     "watch": "concurrently \"tsc -w\" \"nodemon --watch ./build/ --exec npm run start\"",
-    "package": "npm run zip"
+    "package": "npm run zip",
+    "loadEmulatorData": "node ./firestoreEmulator/loadEmulatorData.js",
+    "integrationTest": "firebase emulators:exec --project=test --ui 'npm run loadEmulatorData && npm run test:integration'",
```
Comment (aalmansa-e2x) [inline package.json:27]: maybe prepareIntegrationTests?

**Review by aalmansa-e2x — DISMISSED** [review]

File: .github/workflows/main.yml (hunk 1 above)
Comment (hamzalatif-e2x) [reply .github/workflows/main.yml:42]: yes! if we find a way to start up all our functions with one command then we most likely will only need one command to run integration tests

File: firestoreEmulator/teacher1.json (hunk 2 above)
Comment (hamzalatif-e2x) [reply firestoreEmulator/teacher1.json:5]: yup! happy to change this to categories. We can figure out what that structure looks like and put it here

File: src/index.ts (hunk 3 above)
Comment (hamzalatif-e2x) [reply src/index.ts:17]: yes, just put this in as an example to get everything working

File: .firebaserc (hunk 4 above)
Comment (hamzalatif-e2x) [reply .firebaserc:1]: it was generated when i initialised the firebase, i will test without it and see if we can remove it

**Review by aalmansa-e2x — APPROVED** [review]

**Review by aalmansa-e2x — APPROVED** [review]
